test_communication/package.xml

```diff
@@ -11,21 +11,24 @@
   <maintainer email="dthomas@osrfoundation.org">Dirk Thomas</maintainer>
   <license>Apache License 2.0</license>
 
+  <build_depend>ament_cmake_auto</build_depend>
+  <build_depend>builtin_interfaces</build_depend>
+  <build_depend>rosidl_default_generators</build_depend>
+
   <buildtool_depend>ament_cmake</buildtool_depend>
```

Comment:
What difference in behavior should I expect by calling `ament_auto_package()`? I'm not doubting the advice, but trying to understand why each thing is used in a different context. That's especially difficult to determine for these kinds of  changes to the `package.xml` and `CMakeLists.txt` because everything seems to work fine with a wide variety of different combinations and I don't have the context to identify potential problems by inspection.